src/sshConfig.ts

```diff
@@ -123,14 +124,29 @@ export class SSHConfig {
    */
   private getBlock(label: string): Block | undefined {
     const raw = this.getRaw()
-    const startBlockIndex = raw.indexOf(this.startBlockComment(label))
-    const endBlockIndex = raw.indexOf(this.endBlockComment(label))
+    const startBlock = this.startBlockComment(label)
+    const endBlock = this.endBlockComment(label)
+    const startBlockCount = countSubstring(startBlock, raw)
+    const endBlockCount = countSubstring(endBlock, raw)
+    const startBlockIndex = raw.indexOf(startBlock)
+    const endBlockIndex = raw.indexOf(endBlock)
     const hasBlock = startBlockIndex > -1 && endBlockIndex > -1
 
     if (!hasBlock) {
       return
     }
 
+    if (startBlockCount !== endBlockCount) {
+      throw new SSHConfigBadFormat(
+        `Malformed config: ssh config has ${startBlockCount} ${label} START comments but ${endBlockCount} ${label} END comments. Each START must have a matching END.`,
+      )
+    }
+    if (startBlockCount > 1 || endBlockCount > 1) {
+      throw new SSHConfigBadFormat(
+        `Malformed config: ssh config has ${startBlockCount} ${label} sections, please remove all but one.`,
```

Comment:
I wonder if we should just say `${label} sections` without the `startBlockCount`, or the max of the start and end, or write both counts like the above error, since for example if there are zero start blocks but 1+ end blocks it would say "zero sections, please remove".

Reply:
Won't `startBlockCount` and `endBlockCount` be different then, meaning we won't even hit this code path in that case? You'll get an error like

```
Malformed config: ssh config has 87 dev.coder.com sections, please remove all but one.
```